toolkit/tools/pkg/imagecustomizerlib/customizeuki.go

```diff
@@ -0,0 +1,476 @@
+
+	// Create the INI file.
+	cfg := ini.Empty()
+	section, err := cfg.NewSection("UKI")
+	if err != nil {
+		return fmt.Errorf("failed to create INI section:\n%w", err)
+	}
+
+	// Add keys to the INI file.
+	_, err = section.NewKey("Linux", filepath.Join(buildDir, UkiBuildDir, kernel))
+	if err != nil {
+		return fmt.Errorf("failed to add 'Linux' key to INI file:\n%w", err)
+	}
+
+	_, err = section.NewKey("Initrd", filepath.Join(buildDir, UkiBuildDir, initramfs))
+	if err != nil {
+		return fmt.Errorf("failed to add 'Initrd' key to INI file:\n%w", err)
+	}
+
+	_, err = section.NewKey("Cmdline", fmt.Sprintf("@%s", cmdlineFilePath))
+	if err != nil {
+		return fmt.Errorf("failed to add 'Cmdline' key to INI file:\n%w", err)
+	}
+
+	_, err = section.NewKey("OSRelease", fmt.Sprintf("@%s", osSubreleaseFullPath))
+	if err != nil {
+		return fmt.Errorf("failed to add 'OSRelease' key to INI file:\n%w", err)
+	}
+
+	// Save the INI file.
+	err = cfg.SaveTo(configFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to save INI file for kernel (%s):\n%w", kernelVersion, err)
+	}
+
+	ukiFullPath := filepath.Join(systemBootPartitionTmpDir, UkiOutputDir, fmt.Sprintf("%s.unsigned.efi", kernel))
+
+	// Build the UKI using ukify.
+	ukifyCmd := []string{
+		"-c", configFilePath, "build",
+		fmt.Sprintf("--stub=%s", stubPath),
+		fmt.Sprintf("--output=%s", ukiFullPath),
+	}
+
+	err = shell.ExecuteLiveWithErr(1, "ukify", ukifyCmd...)
+	if err != nil {
+		return fmt.Errorf("failed to build UKI for config (%s):\n%w", configFilePath, err)
+	}
+
+	logger.Log.Infof("Successfully built UKI: (%s)", ukiFullPath)
+	return nil
+}
+
+func cleanupUkiBuildDir(buildDir string) error {
+	ukiBuildDirPath := filepath.Join(buildDir, UkiBuildDir)
+
+	err := os.RemoveAll(ukiBuildDirPath)
+	if err != nil {
+		return fmt.Errorf("failed to clean up UkiBuildDir at (%s):\n%w", ukiBuildDirPath, err)
+	}
+
+	logger.Log.Infof("Successfully cleaned up UkiBuildDir: (%s)", ukiBuildDirPath)
+	return nil
+}
+
+func cleanupBootPartition(bootPartitionTmpDir string) error {
+	dirEntries, err := os.ReadDir(bootPartitionTmpDir)
+	if err != nil {
+		return fmt.Errorf("failed to read boot partition directory (%s):\n%w", bootPartitionTmpDir, err)
+	}
+
+	for _, entry := range dirEntries {
+		entryPath := filepath.Join(bootPartitionTmpDir, entry.Name())
+
+		err := os.RemoveAll(entryPath)
+		if err != nil {
+			return fmt.Errorf("failed to remove (%s):\n%w", entryPath, err)
+		}
+	}
+
+	logger.Log.Infof("Successfully cleaned up boot partition: (%s)", bootPartitionTmpDir)
+	return nil
+}
+
+func appendKernelArgsToUkiCmdlineFile(buildDir string, newArgs []string) error {
+	cmdlineFilePath := filepath.Join(buildDir, UkiBuildDir, "kernel-cmdline-args")
+
+	existingArgs, err := file.Read(cmdlineFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to read kernel cmdline args file at (%s):\n%w", cmdlineFilePath, err)
+	}
+
+	existingArgsSlice := strings.Fields(string(existingArgs))
```

Comment:
The parsing rules are far more complex than this. I would just do: `updatedArgs  = existingArgs + " " + newArgs`